eventsourced/src/lib.rs

```diff
@@ -260,77 +316,112 @@ pub enum SpawnError {
     NextEvt(#[source] BoxError),
 }
 
-impl<E> EventSourcedExt for E where E: EventSourced {}
-
-/// A handle for a spawned event sourced entity which can be used to invoke its command handler.
-#[derive(Debug, Clone)]
-#[allow(clippy::type_complexity)]
-pub struct EntityRef<E>
-where
-    E: EventSourced,
-{
-    cmd_in: mpsc::Sender<(E::Cmd, oneshot::Sender<Result<(), E::Error>>)>,
+pub struct EntityRef<I, E, S, T, Tail> {
```

Comment:
Will these types never end up in user code? How do you pass around these refs if you cannot reasonably type them out any more?

Reply:
`EntityRef<Uuid, Evt, Counter, Decrease, Cmds<Increase, ()>` is how this type looks like in the [example](https://github.com/hseeberger/eventsourced/pull/192/files#diff-d0314e7613a6617bbacdb0c2640c87c54b5914888e35f3827669df4d1468d7c8R33). I agree that this is ugly, but users can make a type alias once. And I plan to create a type macro which can be used to ease making the type alias, something like:

```rust
type CounterEntityRef = EntityRef!(Counter, Uuid, Evt, Increase, Decrease)
```

Reply:
:+1: indeed, that's not too bad, a bit weird how the list of commands needs to be encoded, but maybe ok enough? Is there an implementation of a coproduct in Rust which we could reuse?

Reply:
Neat, will give it a try.

Reply:
Looking good.